src/main/java/com/synopsys/integration/detect/configuration/DetectableOptionFactory.java

```diff
@@ -77,10 +73,8 @@ public DetectableOptionFactory(PropertyConfiguration detectConfiguration, @Nulla
     public BazelDetectableOptions createBazelDetectableOptions() {
         String targetName = getNullableValue(DetectProperties.DETECT_BAZEL_TARGET);
         List<String> bazelCqueryAdditionalOptions = getValue(DetectProperties.DETECT_BAZEL_CQUERY_OPTIONS);
-
-        List<FilterableEnumValue<WorkspaceRule>> bazelDependencyRulesPropertyValues = getValue(DetectProperties.DETECT_BAZEL_DEPENDENCY_RULE);
-        Set<WorkspaceRule> bazelDependencyRules = deriveBazelDependencyRules(bazelDependencyRulesPropertyValues);
-        return new BazelDetectableOptions(targetName, bazelDependencyRules, bazelCqueryAdditionalOptions);
+        List<WorkspaceRule> bazelDependencyRules = getValue(DetectProperties.DETECT_BAZEL_DEPENDENCY_RULE).representedValues();
+        return new BazelDetectableOptions(targetName, new HashSet<>(bazelDependencyRules), bazelCqueryAdditionalOptions);
```

Comment:
This line has me questioning whether 'FilterableEnumList' should be 'FilterableEnumSet'